src/shared/sidepanel/types.ts

```diff
@@ -0,0 +1,21 @@
+import { isObj } from '../isObj';
+
+interface IsOpenRequest {
+  method: 'is-sidepanel-open';
+  params: { windowId: number | null };
+}
+
+interface CloseSidepanelRequest {
+  method: 'close-sidepanel';
+  params: { windowId: number | null };
+}
+
+export type SidepanelMessageRequest = {
+  payload: IsOpenRequest | CloseSidepanelRequest;
+};
+
+export function isSidepanelMessageRequest(
+  x: unknown
+): x is SidepanelMessageRequest {
+  return isObj(x) && isObj(x.payload) && 'method' in x.payload;
```

Comment:
should we check here that x.payload.method is one of options 'is-sidepanel-open'/'close-sidepanel'?

Reply:
This would make it harder to add more methods
Good case for `zod` btw